### PR TITLE
[New] Add jshashtable w/ git auto-update

### PR DESCRIPTION
**Pull request by sashberd**
Pull request for issue: #5985
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/timdown/jshashtable
 * Official website (optional, not the repository): https://www.timdown.co.uk/jshashtable/index.html
 * NPM package url (optional):
 * License and its reference: Apache-2.0
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 7
   - Count of stars: 56
   - Count of forks: 25
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/jshashtable.js/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/timdown/jshashtable.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "hashset.js",
+          "hashtable.js"
```
Comment (sufuf3) [inline ajax/libs/jshashtable.js/package.json]: Let's use `+(hashset| hashtable)*.+(js|css|map)`
we can use one line of pattern to get all the files we need. (via  [node-glob](https://github.com/isaacs/node-glob#glob-primer))
What do you think?

Thank you.

**Comment by sashberd** [comment]
@sufuf3 
The PR was updated. The homepga you provided is not working with https, thus I used http.
I use https://libraries.io to find npm packages and review them. If you will search this librirry o this page you will find it with two versions 2.1 and 3.0. 
https://libraries.io/nuget/jshashtable.js/2.1.0
FYR and FYI

**Comment by sashberd** [comment]
@sufuf3 ping

File: ajax/libs/jshashtable.js/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "hashset"
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/timdown/jshashtable.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "+(hashset| hashtable)*.+(js|css|map)"
```
Comment (sufuf3) [inline ajax/libs/jshashtable.js/package.json]: Please remove the space between `|` and `h`.
We don't have to add extra space.
Also this pattern can't get the `hashtable*` files.
Many thanks

**Comment by sashberd** [comment]
@sufuf3 You are right about version2.1, I removed it. The PR was updated FYR

File: ajax/libs/jshashtable.js/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "jshashtable.js",
```
Comment (sufuf3) [inline ajax/libs/jshashtable.js/package.json]: I saw the library name is `jshashtable`, as the GitHub page & lib's website.
Could we as the same name with upstream?
ref: https://github.com/timdown/jshashtable/blob/master/bower.json#L2
What do you think?

Many thanks.

File: ajax/libs/jshashtable.js/3.0/hashset.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
+/**
```
Comment (sufuf3) [inline ajax/libs/jshashtable.js/3.0/hashset.js]: This file's md5 hash is `3f5549207b6eb75540787ecf1de294a1`
But the origin hash is `cfe58d496d66dfe97c5af27c2aad331a`
Would you please re-add this file and generate the minified files again?
Thank you.

File: ajax/libs/jshashtable.js/3.0/hashtable.js [hunk 5]
```diff
@@ -0,0 +1,402 @@
+/**
```
Comment (sufuf3) [inline ajax/libs/jshashtable.js/3.0/hashtable.js]: This file's md5 hash is `041054fb273de9c0ecdbaeef262050d9`
But the origin hash is `b2d65ae2edd7a7f7c751b2f02f76fb1e`
Would you please re-add this file and generate the minified files again?
Thank you.

**Comment by sashberd** [comment]
@sufuf3 The PR was updated. FYR

**Comment by sufuf3** [comment]
Would you please also modify the commit message(The lib's name)?
Many thanks.

**Comment by PeterDaveHello** [comment]
Add it if it comes from the reliable source, do not add it in the other side.

**Comment by extend1994** [comment]
@sashberd Would you like to add old versions by tracing commit logs?

**Comment by sashberd** [comment]
@extend1994 The request of this library was more then  years, moreover, the library is not maintained today, and the latest version is good enough i think. 
Nevertheless, what kind of version you are talking about? I did not saw any old exact  versions in log, only like "first version",what is version number,  or  "completed version for version 2", is it v2.0 or something else... it is not clear and I think we will not get answer from the author

**Comment by extend1994** [comment]
@sashberd 
v1
https://github.com/timdown/jshashtable/commit/c116908ab81c9316464e23a87ffcdbc95938ed9a#diff-0fa7f3b1222a93260b715e8b9abbf58cR24
v2
https://github.com/timdown/jshashtable/commit/abf796bc04cd6d1ff212cce912206407bd3a57da
v2.1
https://github.com/timdown/jshashtable/commit/3552984182a511f23d28bfb1623771ead1947796
v2.2
https://github.com/timdown/jshashtable/commit/7c864988a43f32c50cde1aef5be04e3b108b6c19

**Comment by sashberd** [comment]
@extend1994  The PR was updated. FYR

**Comment by sashberd** [comment]
@extend1994 The PR was updated. I used wget in order to get all those files from the source and I did not touch them ever. I already faced with this problem with @sufuf3 and I think that is because I am using different OS then yours. The only solution is to add those files from you OS.

**Comment by sufuf3** [comment]
Ummm... What I found is not in this case.

There are different `md5sum` between the files which were fetched by you and me.
Because I used `git diff` & `diff` and found that the files are not be touched but they have the different checksum. This case is a little bit weird.

But in this case, the file is different via using `git diff`.
So, maybe tell us more detail is good to discuss.
What do you think?

Many thanks.

**Comment by sashberd** [comment]
@extend1994 The files of version 3.0 were readed with wget tool . FYR

**Comment by sufuf3** [comment]
@sashberd Thanks for the PR.
Would you please also mention the commit hash value in commit message?
Because when we need to fine the version was add from which commit, we will have the reference from commit message. This is really useful for us.
What do you think?
Many thanks.

PS. like mention
```
v1 timdown/jshashtable@c116908#diff-0fa7f3b1222a93260b715e8b9abbf58cR24
v2 timdown/jshashtable@abf796b
v2.1 timdown/jshashtable@3552984
v2.2 timdown/jshashtable@7c86498
```
in second commit message.


Thank you very much.

**Comment by sufuf3** [comment]
The first commit message should be modified the word from `minifing` to `minifying`, be added a `y` between `f` & `i`.
Thanks.

**Comment by sashberd** [comment]
@sufuf3 The PR was updated. FYR

**Comment by sashberd** [comment]
@sufuf3 ping

**Comment by extend1994** [comment]
@sashberd Would you please try https://help.github.com/articles/dealing-with-line-endings/#platform-windows? Let's see if it's fixed in this way. Thanks.

**Comment by sashberd** [comment]
@extend1994 For what goal I should use this? 

> Let's see if it's fixed in this way
What it should be fix?

**Comment by extend1994** [comment]
@sashberd https://github.com/cdnjs/cdnjs/pull/12042#pullrequestreview-91151854
Line-endings issue.

**Comment by sashberd** [comment]
@extend1994 The PR was updated. FYR

**Comment by sashberd** [comment]
@extend1994 The v3.0 files were saved by UNIX scheme. FYR if such solution helps to solve the problem

**Comment by sufuf3** [comment]
Would you revise the second commit message from `using web-minify-tool for minifng files` to `using web-minify-tool for minifying files` (`minifng` -> `minifying`)?
Many thanks.

**Comment by sashberd** [comment]
@sufuf3 Updated

**Comment by extend1994** [comment]
@sashberd Did you re-add and re-commit `hashset.js` & `hashtable.js`?

**Comment by sashberd** [comment]
@extend1994 I did it before and I re add them in this commit . If still there is ^M char I have no way to fix it only you from your side(Linux) can fix it

**Comment by PeterDaveHello** [comment]
@sashberd hmmm, I believe that's something can be fixed, do you have git config about the file format? about the clrf.

**Comment by sashberd** [comment]
@PeterDaveHello Are you talking about https://help.github.com/articles/dealing-with-line-endings/?

**Comment by PeterDaveHello** [comment]
@sashberd yes, you are right!

**Comment by sashberd** [comment]
@PeterDaveHello I will set it to true and readd all v3.0

**Comment by PeterDaveHello** [comment]
@sashberd I thought it should be false for all assets we added, we don't want to touch them.

**Comment by sashberd** [comment]
@PeterDaveHello OK it was on false in previous comment. Thus, if the problem still exists I want to ask you to help me with it cause we will never exit from this deadlock

**Comment by PeterDaveHello** [comment]
@sashberd this is not a deadlock, there must be a way to figure it out, just need few more steps and minutes to solved it, right?

**Comment by sashberd** [comment]
@PeterDaveHello Is issue still exists in last commit?

**Comment by sashberd** [comment]
@sufuf3 @extend1994 The PR was updated. FYR

**Comment by sashberd** [comment]
@sufuf3 @extend1994 The PR was updated with v3.0 from Mac. I think this one will fix problem with ^M in files

**Comment by sashberd** [comment]
@extend1994 @sufuf3 ping

**Comment by sashberd** [comment]
@sufuf3 @extend1994 ping

**Comment by sashberd** [comment]
@extend1994 The PR was updated. FYR
@PeterDaveHello @sufuf3 @extend1994 I cannot fix the problem with newline. I have no idea why it si there. Please help me to fix it in order to finish this PR

**Comment by sashberd** [comment]
@xtuc @MattIPv4 This one has long story and conversation. Here also we had add versions manually from commit history, but the lib add was connected to https://www.nuget.org/packages/jshashtable.js counter and not GitHub counters, which a low. Thus, under you consideration guys what todo with it.

**Pull request closed without merging**